packages/material-ui-lab/src/TreeView/TreeView.js

```diff
@@ -80,11 +68,18 @@ const TreeView = React.forwardRef(function TreeView(props, ref) {
     state: 'selected',
   });
 
-  const getChildren = (id) =>
-    objectValues(nodeMap.current)
-      .filter((node) => node.parentId === id)
-      .sort((a, b) => a.index - b.index)
-      .map((child) => child.id);
+  const getChildrenIds = (id) => {
+    const childIds = [];
+
+    Object.keys(nodeMap.current).forEach((key) => {
+      const value = nodeMap.current[key];
+      if (value.parentId === id) {
+        childIds.splice(value.index, 0, value.id);
```

Comment:
Doesn't this produce an array with holes? I feel like the previous .filter + .sort was more readable and creates a more "standard" array.

Reply:
Erm, the previous was probably more readable. This shouldn't create holes since the index is updated and should cover all the indexes.